src/parse.rs

```diff
@@ -40,6 +40,39 @@ pub enum Pattern {
     Product(Arc<Self>, Arc<Self>),
 }
 
+impl Pattern {
+    /// Create an equivalent pattern that corresponds to the Simplicity base types.
```

Comment:
3d65d4b341b628a4d46dfb280b29551be1ca9a92: I'm not sure if "structural types" would be a better name. Arguably, values of a type always follow the type's structure.

We are trying to describe the types that are part of the definition of the Simplicity type system. These types are not defined in terms of other types.

Reply:
Ideally we choose a name that fits both the Simplicity side and the Simfony side.